### PR TITLE
Config changes

### PR DESCRIPTION
**Pull request by wxtim**
Change a small number of links referring to config items which have been moved/renamed.

Partially closes_: https://github.com/cylc/cylc-flow/issues/3696

### CODE REVIEW

File: cylc/doc/etc/tutorial/runtime-introduction/.validate [hunk 1]
```diff
@@ -21,7 +21,6 @@ set -eux
 TEMPORARYDIRECTORY=$(mktemp -dp "$HOME"/cylc-run/)
 cp -r ./* "$TEMPORARYDIRECTORY"
 cd "$TEMPORARYDIRECTORY"
-sed -i '1s;^;[scheduler]\n    abort if any task fails = True\n;' flow.cylc
```
Comment (MetRonnie) [inline cylc/doc/etc/tutorial/runtime-introduction/.validate:24]: Yuk 😆 A lot nicer now

File: src/user-guide/running-suites.rst [hunk 2]
```diff
@@ -644,7 +644,6 @@ These can be configured using the settings:
 
 - :cylc:conf:`[mail]to` (list of email addresses)
 - :cylc:conf:`[mail]from`
-- :cylc:conf:`[mail]smtp`
```
Comment (MetRonnie) [inline src/user-guide/running-suites.rst:647]: The SMTP server can still be set, but only in global config. Maybe it's worth linking to that?

Comment (wxtim) [reply src/user-guide/running-suites.rst:647]: I don't think so - I think that we'd rather most users don't tinker with the global config.

**Review by MetRonnie — APPROVED** [review]

File: src/user-guide/running-suites.rst (hunk 2 above)
Comment (MetRonnie) [reply src/user-guide/running-suites.rst:647]: Ok fair enough

File: cylc/doc/etc/tutorial/runtime-introduction/.validate (hunk 1 above)
Comment (wxtim) [reply cylc/doc/etc/tutorial/runtime-introduction/.validate:24]: It took me a while and some panicky message to Oliver to sort this. It's an important lesson with Sed though: Separators can be chars not="/", and it's often convenient if they are something else

**Review by kinow — APPROVED** [review]

**Pull request merged**